### DISCUSSION TITLE
on("DELETE", ...).subscribe();

### DISCUSSION THREAD

**Discussion by tklein1801** [supabase/supabase#1904] [Questions]
Hey, I have a little problem using real-time subscriptions for DELETE-events.
If I use this code
```js
supbase
  .from(`post_comments:post_id=eq.${postData.id}`)
  .on("DELETE", (payload) => {
     // Do stuff
  })
  .subscribe();
```

What I'm trying to is I wanna listen if the an comment from this post get deleted so I can remove it from the current comment-list so it won't displayed anymore.
My problem is that my subscription only receives an object containing `{comment_id: 50, post_id: null, user_id: null, etc...}` so my subscription won't get triggered because post_id equals null and not the real post id.

So maybe one of you can explain me my mistake and tell how I can fix this.

**Comment by cdedreuille** [comment]
You can try perhaps to listen to the full table instead and filter your list of comments using the `old` id.
Something like this (untested):
```
supbase
  .from('post_comments')
  .on("DELETE", (payload) => {
     const newList = listOfcomments.filter(comment => comment.id !== payload.old.id)
     // Do some stuff with this new list like updating your state.
  })
  .subscribe();
```

**Reply by tklein1801** [reply]
I know but why should I try it on this way if there is an better way how to do it? Currently I'm doing it the same way you did in your example but I wanna reduce the payload for the user. If many comments are published more will be deleted...
